com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/MetricsSink.java

```diff
@@ -0,0 +1,362 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingInput;
+import com.ibm.streams.operator.Tuple;
+import com.ibm.streams.operator.model.InputPortSet;
+import com.ibm.streams.operator.model.InputPortSet.WindowMode;
+import com.ibm.streams.operator.model.InputPortSet.WindowPunctuationInputMode;
+import com.ibm.streams.operator.model.InputPorts;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+/**
+ * Class for an operator that consumes tuples. 
+ * The following event methods from the Operator interface can be called:
+ * </p>
+ * <ul>
+ * <li><code>initialize()</code> to perform operator initialization</li>
+ * <li>allPortsReady() notification indicates the operator's ports are ready to process and submit tuples</li> 
+ * <li>process() handles a tuple arriving on an input port 
+ * <li>processPuncuation() handles a punctuation mark arriving on an input port 
+ * <li>shutdown() to shutdown the operator. A shutdown request may occur at any time, 
+ * such as a request to stop a PE or cancel a job. 
+ * Thus the shutdown() may occur while the operator is processing tuples, punctuation marks, 
+ * or even during port ready notification.</li>
+ * </ul>
+ * <p>With the exception of operator initialization, all the other events may occur concurrently with each other, 
+ * which lead to these methods being called concurrently by different threads.</p> 
+ */
+@PrimitiveOperator(
+		name="MetricsSink",
+		namespace="com.ibm.streamsx.metrics",
+		description=MetricsSource.DESC_OPERATOR
+		)
+@InputPorts({
+	@InputPortSet(
+			description="Port that ingests tuples",
+			cardinality=1,
+			optional=false,
+			windowingMode=WindowMode.NonWindowed,
+			windowPunctuationInputMode=WindowPunctuationInputMode.Oblivious
+			),
+	@InputPortSet(
+			description="Optional input ports",
+			optional=true,
+			windowingMode=WindowMode.NonWindowed,
+			windowPunctuationInputMode=WindowPunctuationInputMode.Oblivious
+			)
+})
+@Libraries({
+	"opt/downloaded/*"
+	})
+public class MetricsSink extends AbstractOperator {
```

Comment:
Yes, I talked with Pete and he said it was his intention all along to split the operators. I'll stick with "MetricsElasticSearchSink" and "MetricsInfluxDBSink" for now and can change it later on if we decide on better names.

And yes, I can make a more generic approach for outputting the complete tuple to the underlying databases. For the metricName-metricValue pair, I'll just output the metrics as-is without joining them. This just means that querying for metric values in the databases would go from looking like this:
`SELECT "<metricName>" FROM "<databaseName>"`

to something more like this:
`SELECT "metricValue" FROM "<databaseName>" WHERE "metricName"="<metricName>"`